### ISSUE TITLE
Parser parses incorrectly in British Summer Time

### ISSUE THREAD

**Issue by Devking** [dateutil/dateutil#370]
**The Problem**
Tests fail in British Summer Time (BST).

**To Reproduce**
Set timezone on your computer to BST and run tests. This is confirmed to happen on both MacOS and Ubuntu, on both Python2 and Python3.

**Investigation**

On the Python interpreter in any timezone that is not BST (nor GMT):

```
>>> import dateutil.parser
>>> date = 'Fri, 14 Apr 2017 16:18:53 GMT'
>>> dateutil.parser.parse(date)
datetime.datetime(2017, 4, 14, 16, 18, 53, tzinfo=tzutc())
```

GMT parses to `tzutc()` as expected.

In BST and GMT:

```
>>> import dateutil.parser
>>> date = 'Fri, 14 Apr 2017 16:18:53 GMT'
>>> dateutil.parser.parse(date)
datetime.datetime(2017, 4, 14, 16, 18, 53, tzinfo=tzlocal())
```

GMT now parses to `tzlocal()`, which is not correct if you're in BST, since BST is one hour off of GMT.

**Comment by pganssle** [comment]
Duplicate of #349.

**Comment by pganssle** [comment]
Reopening this, because I think it's actually a duplicate of #356 and this one has more information (#349 is a similar, related bug, but I think caused by a slightly different code path).

**Comment by pganssle** [comment]
Some commentary from #349 that's actually probably more appropriate here:

>This happens because `dateutil` tries to make some context-dependent guesses about what time zone you mean when the time zone is ambiguous. `GMT` could be `Europe/London` or `UTC`, and apparently it prioritizes `local()` over any other interpretation. I personally don't really like this implicit context, but it's what we have.

> A somewhat related bug is #320. I think both can be fixed with a "sanity check" when assigning time zones to parsed dates, where, when you are pulling something that's supposed to be a `tzname()` output, you make sure that that `tzname()` *actually* applies at the time of year you've parsed.

> Currently I think the best workaround would be to replace "GMT" with "UTC", which is hopefully unambiguous.

**Comment by Devking** [comment]
Yup, saw your comment there. I can also confirm that parsing a date that specifies `UTC` instead of `GMT` will avoid this ambiguity. Thanks for the quick investigation!

**Comment by pganssle** [comment]
This is partially fixed by #517, though the issue from #349 is still active.

Another fix for this is to explicitly pass a `tzinfos` dict that contains `{x: tz.tzutc() for x in ['UTC', 'GMT', 'Z']}`, as that will be prioritized over the local zone detection.

**Comment by joncasdam** [comment]
I would say this is still going on.

```
foo = parse('25/06/18 13:53:33 UTC+1', parserinfo=parserinfo(dayfirst=True))
# if we check foo
foo
datetime.datetime(2018, 6, 25, 13, 53, 33, tzinfo=tzoffset(None, -3600))
# if we print foo
print(foo)
2018-06-25 13:53:33-01:00
# I'm in Brussels time zone, which now means CEST)
foo.astimezone(pytz.timezone('Europe/Brussels'))
datetime.datetime(2018, 6, 25, 16, 53, 33, tzinfo=<DstTzInfo 'Europe/Brussels' CEST+2:00:00 DST>)
```
can you see the 3 hours jump?

so I think it's not only related to BST. maybe something in the correlation of UTC timetable and the other timetable (GMT, BST, CET, CEST and so on)

**Comment by pganssle** [comment]
@joncasdam I am fairly certain that is a different issue. `UTC+1` is the same thing as `-01:00` because it seems to be a POSIX-style datetime, so it's not surprising that converting from `-01:00` to `+02:00` would be 3 hours different.

As an aside, [I would recommend against using pytz](https://blog.ganssle.io/articles/2018/03/pytz-fastest-footgun.html).
